src/algorithm/solarPanelPlacement.ts

```diff
@@ -0,0 +1,259 @@
+import {
+  Feature,
+  FeatureCollection,
+  Geometry,
+  Point,
+  Polygon,
+  Properties,
+  featureCollection,
+  point,
+  polygon,
+} from '@turf/turf'
+
+import * as fs from 'fs'
+import * as path from 'path'
+
+export type Square = {
+  usable: boolean
+  squareCenter: Geometry
+}
+
+export type Matrix = Square[][]
+
+export type SolarPanelGrid = {
+  gridIndexes: [number, number][]
+}
+
+// Helper function
+
+export function writeFeature(
+  outputFileName: string,
+  feature: FeatureCollection | Feature,
+  debug: boolean = false
+) {
+  if (debug) {
+    fs.writeFileSync(
+      path.join(__dirname, outputFileName),
+      JSON.stringify(feature)
+    )
+  }
+}
+
+function createSolarPanel(
```

Comment:
Can you add one example with the value of index and horizontal and the result of function please